File: benchmarks/go/file_io.go
package main

import (
	"bytes"
	"os"
)

func main() {
	path := "tn_benchmark_file_io.tmp"
	chunk := []byte("TezzNativeBenchmarkBlock")
	reps := 12000
	expected := len(chunk) * reps

	out, err := os.Create(path)
	if err != nil {
		os.Exit(1)
	}
	for i := 0; i < reps; i++ {
		if n, err := out.Write(chunk); err != nil || n != len(chunk) {
			out.Close()
			os.Remove(path)
			os.Exit(2)
		}
	}
	if out.Close() != nil {
		os.Remove(path)
		os.Exit(3)
	}
	info, err := os.Stat(path)
	if err != nil || int(info.Size()) != expected {
		os.Remove(path)
		os.Exit(4)
	}

	data, err := os.ReadFile(path)
	os.Remove(path)
	if err != nil || len(data) != expected || !bytes.Equal(data[:len(chunk)], chunk) {
		os.Exit(5)
	}
	acc := 0
	for _, b := range data {
		acc = (acc + int(b)) & 1048575
	}
	if acc == -1 {
		os.Exit(6)
	}
}

File: benchmarks/python/file_io.py
from pathlib import Path

path = Path("tn_benchmark_file_io.tmp")
chunk = b"TezzNativeBenchmarkBlock"
reps = 12_000
expected = len(chunk) * reps

with path.open("wb") as f:
    for _ in range(reps):
        f.write(chunk)

if path.stat().st_size != expected:
    path.unlink(missing_ok=True)
    raise SystemExit(1)

total = 0
acc = 0
with path.open("rb") as f:
    while total < expected:
        data = f.read(len(chunk))
        if len(data) != len(chunk):
            path.unlink(missing_ok=True)
            raise SystemExit(2)
        for b in data:
            acc = (acc + b) & 1_048_575
        total += len(data)

path.unlink(missing_ok=True)
if total != expected or acc == -1:
    raise SystemExit(3)
